index.js

```diff
@@ -17,19 +17,52 @@ app.use(express.static("public"));
 
 io.on("connection", (socket) => {
   // hash is the geohash of the user's location
-  const { hash } = socket.handshake.query;
+  let { hash, username } = socket.handshake.query;
 
   if (hash) {
     // Join the user to the grid
     socket.join(hash);
 
     if (grids[hash]) {
+      grids[hash].add({
```

Comment:
Only save the username in this set, you already know the everyone in this set belongs to the region with hash`xyx`, so then storing `xyz` again for every user is redundant. Doing the current way also means you aren't using any of the benefits of a set, since unlike storing a primitive value like string, every object you make will be unique for the set even if the username and hash inside it is the same. So you are making a set but always using it as an array, which is not efficient. If you only store the username you can then perform O(1) deletion later, instead of converting it into an array and doing a linear search which negates the whole benefit of a set.